src/Processors/Formats/Impl/ParallelParsingInputFormat.cpp

```diff
@@ -137,7 +137,8 @@ void ParallelParsingInputFormat::onBackgroundException(size_t offset)
             if (e->getLineNumber() != -1)
                 e->setLineNumber(e->getLineNumber() + offset);
     }
-    tryLogCurrentException(__PRETTY_FUNCTION__);
+    if (is_server)
+        tryLogCurrentException(__PRETTY_FUNCTION__);
```

Comment:
I checked and saw that second exception is actually printed here
https://github.com/ClickHouse/ClickHouse/blob/b4e30ad78ef1bd16caec7b10adeb8077f7b2b46a/src/Client/ClientBase.cpp#L1562-L1567 i.e. looks like this `is_server` flag is equal to just removing this line with exception printing completely.
So may be just remove it then? I did that locally and there is indeed only one exception as needed.

Reply:
This exception printing in ClientBase is the original place where exception in client should be printed. In my example `insert into test_table format TSV some trash here 123 \n 456` the first exception with stacktrace from ParallelParsingInputFormat shouldn't be printed (it happens because of calling `tryLogCurrentException`, which make sense only for server, not for client)